CHANGELOG.md

```diff
@@ -4,6 +4,375 @@ All notable changes to Router will be documented in this file.
+
+This feature is explicitly designed to:
+
+"""
+Use sufficient precision when parsing fixed precision floats from JSON to ensure that they maintain accuracy when round-tripped through JSON. This comes at an approximately 2x performance cost for parsing floats compared to the default best-effort precision.
+"""
+
+Despite the performance impact for floats, we need the fix in order to prevent float values losing precision when processed by the router.
+
+By [@garypen](https://github.com/garypen) in https://github.com/apollographql/router/pull/3338
+
+### Fix @defer inside an aliased field ([Issue #3263](https://github.com/apollographql/router/issues/3263))
+
+This refactors [PR #3298](https://github.com/apollographql/router/pull/3298/) to prepare deferred subselections without serializing them to GraphQL syntax just to re-parse them. This removes code that did not correctly handle some interactions of `@defer` with aliases.
+
+By [@SimonSapin](https://github.com/SimonSapin) in https://github.com/apollographql/router/pull/3346
+
+### Fix deferred response formatting when filtering queries ([PR #3298](https://github.com/apollographql/router/pull/3298), [PR #3339](https://github.com/apollographql/router/pull/3339))
+
+Filtering queries requires two levels of response formatting, and its implementation shone light on issues with deferred responses. The response formatting side needs to recognize which deferred fragment generated it, and the deferred response shapes can change depending on request variables, due to the `@defer` directive's `if` argument. So far the response formatting code was not handling those variables.
+For now, this is solved by generating the response shapes for primary and deferred responses, for each combination of the variables used in `@defer` applications, limited to 32 unique variables. There will be follow up work with another approach that removes this limitation, but it will require a lot of deep changes in response formatting, so it will have to land a bit later.
+
+By [@Geal](https://github.com/Geal) in https://github.com/apollographql/router/pull/3298 and https://github.com/apollographql/router/pull/3339
+
+### Otel Ensure that service name is correctly picked up from env and resources ([Issue #3215](https://github.com/apollographql/router/issues/3215))
+
+`OTEL_SERVICE_NAME` env and `service.name` resource are now correctly used when creating tracing exporters.
+
+By [@BrynCooke](https://github.com/BrynCooke) in https://github.com/apollographql/router/pull/3307
+
+## 🛠 Maintenance
+
+### use a Drop guard to track active requests ([PR #3343](https://github.com/apollographql/router/pull/3343))
+
+manually tracking active requests is error prone because we might return early without decrementing the active requests. To make sure this is done properly, `enter_active_request` now returns a guard struct, that will decrement the count on drop
```

Comment:
```suggestion
Manually tracking active requests is error prone because we might return early without decrementing the active requests. To make sure this is done properly, `enter_active_request` now returns a guard struct, that will decrement the count on drop
```